clients/venice-push-job/src/main/java/com/linkedin/venice/hadoop/ValidateSchemaAndBuildDictOutputFormat.java

```diff
@@ -77,15 +72,15 @@ private static void createAndSetDirectoryPermission(
    */
   protected static void setValidateSchemaAndBuildDictionaryOutputDirPath(JobConf job) throws IOException {
     // parent directory
+    FileSystem fs = FileSystem.get(job);
     String parentOutputDir = job.get(MAPPER_OUTPUT_DIRECTORY);
     Path outputPath = new Path(parentOutputDir);
-    FileSystem fs = outputPath.getFileSystem(job);
-    createAndSetDirectoryPermission(fs, outputPath, "777");
+    createDirectoryWithPermission(fs, outputPath, "777");
 
     // store specific directory under parent directory: already derived in VPJ driver and passed along.
     String fullOutputDir = job.get(VALIDATE_SCHEMA_AND_BUILD_DICT_MAPPER_OUTPUT_DIRECTORY);
     outputPath = new Path(fullOutputDir);
-    createAndSetDirectoryPermission(fs, outputPath, "700");
+    createDirectoryWithPermission(fs, outputPath, "700");
```

Comment:
It just occurred to me that we might need to handle multiple headless accounts writing to the same store

Reply:
Does it make sense to have `777` for the store directory too and only set `700` for the job-specific directory?

Reply:
updated the comments here. This is a unique named directory in the format`{$storeName}-{$JOB_EXEC_ID}-{$randomUniqueString}` and will be trashed after this. So won't be used by other accounts.